.github/workflows/test.yml

```diff
@@ -6,138 +6,22 @@ on:
-            django: "Django>=3.2,<4.0"
-            wagtail: "wagtail>=5.2,<5.3"
-            database: "postgresql"
-            experimental: false
-          - python: "3.9"
-            django: "Django>=4.1,<4.2"
-            wagtail: "wagtail>=5.2,<5.3"
-            database: "sqlite3"
-            experimental: false
-          - python: "3.10"
-            django: "Django>=3.2,<4.0"
-            wagtail: "wagtail>=4.1,<4.2"
-            database: "postgresql"
-            experimental: false
-          - python: "3.10"
-            django: "Django>=4.1,<4.2"
-            wagtail: "wagtail>=4.1,<4.2"
-            database: "sqlite3"
-            experimental: false
-          - python: "3.10"
-            django: "Django>=3.2,<4.0"
-            wagtail: "wagtail>=5.2,<5.3"
-            database: "postgresql"
-            experimental: false
-          - python: "3.10"
-            django: "Django>=4.1,<4.2"
-            wagtail: "wagtail>=5.2,<5.3"
-            database: "sqlite3"
-            experimental: false
-          - python: "3.11"
-            django: "Django>=4.1,<4.2"
-            wagtail: "wagtail>=4.1,<4.2"
-            database: "sqlite3"
-            experimental: false
-          - python: "3.11"
-            django: "Django>=4.1,<4.2"
-            wagtail: "wagtail>=5.2,<5.3"
-            database: "sqlite3"
-            experimental: false
-          - python: "3.11"
-            django: "Django>=4.2,<4.3"
-            wagtail: "wagtail>=5.2,<5.3"
-            database: "postgresql"
-            experimental: false
-
-          - python: "3.11"
-            django: "Django>=4.2,<4.3"
-            wagtail: "git+https://github.com/wagtail/wagtail.git@main#egg=wagtail"
-            database: "sqlite3"
-            experimental: true
-
-    services:
-      postgres:
-        image: postgres:14
-        env:
-          POSTGRES_PASSWORD: postgres
-        ports:
-          - 5432:5432
-        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
 
     steps:
       - uses: actions/checkout@v3
-      - name: Set up Python ${{ matrix.python }}
+      - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v4
         with:
-          python-version: ${{ matrix.python }}
+          python-version: ${{ matrix.python-version }}
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install "psycopg2>=2.9.3"
-          pip install "${{ matrix.django }}"
-          pip install "${{ matrix.wagtail }}"
-          pip install -e .[testing]
+          python -m pip install tox tox-gh-actions
       - name: Test
-        run: ./runtests.py
-        env:
-          DATABASE_ENGINE: django.db.backends.${{ matrix.database }}
-          DATABASE_HOST: localhost
-          DATABASE_USER: postgres
-          DATABASE_PASS: postgres
+        run: tox
```

Comment:
This still has the flaw that I raised in https://github.com/torchbox/wagtail-webstories/pull/25#issuecomment-1816674473: for each Python version, we're calling `tox` with no arguments, so it ignores all of the Wagtail / Django version specifiers set up in tox.ini, and just does a single test run with the latest available Wagtail / Django packages for that Python version - which means that nothing below Wagtail 5.2.1 / Django 4.2.7 is ever tested.

You can see this from the output of any of the test runs in https://github.com/torchbox/wagtail-webstories/actions/ - the output of the `tox` command always ends with

```
Successfully installed Django-4.2.7 Pillow-9.5.0 Willow-1.6.2 anyascii-0.3.2 asgiref-3.7.2 beautifulsoup4-4.11.2 black-23.11.0 bleach-3.3.1 certifi-2023.11.17 cfgv-3.4.0 charset-normalizer-3.3.2 click-8.1.7 defusedxml-0.7.1 distlib-0.3.7 django-filter-23.4 django-modelcluster-6.1 django-permissionedforms-0.1 django-taggit-4.0.0 django-treebeard-4.7 djangorestframework-3.14.0 draftjs-exporter-2.1.7 et-xmlfile-1.1.0 filelock-3.13.1 filetype-1.2.0 flake8-6.1.0 html5lib-1.1 identify-2.5.32 idna-3.4 isort-5.12.0 l18n-2021.3 mccabe-0.7.0 mypy-extensions-1.0.0 nodeenv-1.8.0 openpyxl-3.1.2 packaging-23.2 pathspec-0.11.2 pillow-heif-0.13.1 platformdirs-4.0.0 pre-commit-3.5.0 psycopg2-2.9.9 pycodestyle-2.11.1 pyflakes-3.1.0 pytz-2023.3.post1 pyyaml-6.0.1 requests-2.31.0 responses-0.24.1 six-1.16.0 soupsieve-2.5 sqlparse-0.4.4 telepath-0.3.1 tomli-2.0.1 typing-extensions-4.8.0 urllib3-2.1.0 virtualenv-20.24.7 wagtail-5.2.1 wagtail-webstories-0.1 wagtailmedia-0.14.5 webencodings-0.5.1 webstories-0.0.2
```